ci/travis/install-toolchains.sh

```diff
@@ -50,4 +50,16 @@ install_clang() {
   "${cc}" --version
 }
 
-install_clang "$@"
+install_toolchains() {
+  local uses_clang=1 some_lightweight_target="//:sha256"
+  if bazel aquery --noshow_progress --color=no --include_commandline=false --output=textproto --noimplicit_deps "${some_lightweight_target}" |
+     grep -q "^  exec_path: .*/external_Slocal_Uconfig_Ucc_Cmsvc_Ucompiler_Ufiles.*"; then
+    # We detected that we use MSVC, not Clang
+    uses_clang=0
```

Comment:
sorry in previous comment I was talking about not to do this kind of "smart" detection via bazel query. rather, explicitly pass in `install_toochains clang/msvc` as arguments or env var.

Reply:
Oh, I don't think that's a good idea, it's very brittle and actually harder to maintain. There are lots of ways the the compiler can be changed (`--compiler=msvc` on the command line, `--config=msvc` on the command line, `--compiler=msvc` in `.bazelrc`, `--config=msvc` in `.bazelrc`, leaving the default, etc.). They might easily be set differently in different cases, and the only way (and the proper way) to detect e.g. a change in `.bazelrc` is via actually querying Bazel. I think this is the right way to do it. It's not really unmaintainable either IMO–it's just a grep over aquery and it's querying the ground truth.

Reply:
I'll simplify the query options and take out the `exec_path` bit.